torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -3425,6 +3425,28 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::collective(
       !coalescing_state_ && capture_status == c10::cuda::CaptureStatus::None;
   auto work = initWork(
       device, rank_, opType, false, profilingTitle, inputs, outputs, enqueue);
+  if (coalescing_state_) {
+    // When coalescing, we record events per op that lack timing/state
+    // information becuase there is no 'work' associated with them, and then
+    // later in endCoalescing we record a 'coalesced' Work which has
+    // timing/state updates via watchdog thread, but lacks op metadata such as
+    // input/output sizes and profilingTitle per-op in the group.
+    auto trace_id = FlightRecorder::get()->record(
```

Comment:
nit: why store trace_id in a local and then void it?

Reply:
I don't know, tbh, we did that for P2P as well, so we just copy it.. let me try to remove it. I agree we probably don't need it.